### DISCUSSION TITLE
[Proposal] Remove `region` construction from language

### DISCUSSION THREAD

**Discussion by stanislavorlov** [dotnet/csharplang#6996] [Language Ideas] [answered]
Let's remove `region` syntax construction from a c# language since it is making a code no more readable, but instead adding complexity for its readability and file structure.

**Comment by HaloFour** [comment]
The feature cannot be removed without breaking every project that currently uses it.  That seems pretty unnecessary.

I personally like regions for the sake of grouping similar constructs in a class, despite the fact that I don't believe they are generally used by designers for generated code anymore.

**Comment by iam3yal** [comment]
@stanislavorlov Don't use it and make IDEs ignore it this makes a lot more sense than removing it from the language and try to remove it from all codebases out there. :D

**Comment by CyrusNajmabadi** [comment, marked as answer]
We will not do this.  Breaking changes must both have huge value, and barely any customers hit.  Neither holds here.  If you do not want to use `#regions` in your code, then add an analyzer that blocks it for your source code.
